package.json

```diff
@@ -28,21 +28,25 @@
     "rimraf": "^2.6.1",
     "ts-loader": "^2.0.1",
     "typescript": "^2.1.6",
-    "zone.js": ">=0.7.2"
+    "@angular/common": "4.0.2",
+    "@angular/compiler": "4.0.2",
+    "@angular/core": "4.0.2",
+    "@angular/platform-browser": "4.0.2",
+    "@angular/platform-browser-dynamic": "4.0.2",
+    "rxjs": "^5.0.3",
+    "zone.js": ">=0.8.5"
   },
   "dependencies": {
-    "@angular/common": ">=2.0.0",
-    "@angular/compiler": ">=2.0.0",
-    "@angular/core": ">=2.0.0",
-    "@angular/platform-browser": ">=2.0.0",
-    "@angular/platform-browser-dynamic": ">=2.0.0",
     "bootstrap": "^3.3",
     "eonasdan-bootstrap-datetimepicker": ">=4.17.22 <5.0.0",
     "@types/eonasdan-bootstrap-datetimepicker": "^4.17.22",
     "@types/jquery": "^2.0.33",
     "moment": "^2.10",
     "moment-timezone": "^0.4.0",
-    "jquery": "^1.8.3 || ^2.0",
-    "rxjs": "^5.0.1"
+    "jquery": "^1.8.3 || ^2.0"
+  },
+  "peerDependencies": {
```

Comment:
Could you tell me please why is adding peer dependency instead of normal dependency with very low starting version beneficial?

Also as you added angular 4 as dev dependency, is this project still compatible with angular 2?

Reply:
devDependencies is there because we dont need these plugin in the main app, we just need these during building/compiling the script. Before, you put these lib in dependecies and when user install the plugin by npm, user is forced to have all plugin listed in the dependencies in the main app even user dont need these lib or , dont need these lib listed because almost these lib are already  installed with default angular app (and this plugin is for angular so dont need to re-put anguar/core, ...) imo. Since the plugin would work for angular version more  than 2, we dont need to put that in dependecies because user who want to use the plugin should already have angular 2+ in these project, so I think putting it in peerDependencies is good idea by specifiying the minimal version of angular.

The main idea to have devDependecies is because now we can add more lib for building the script, only contributor/Developper of the plugin will install these lib. dependecies are always installed if missing in the main app. peerDependencies will show warning if not matched.

About, Angular 4 in devDependecies, it's always compatible for angular 2, it's just for compiling the script only.

Reply:
OK. You are right. 
I thought maybe we could use some1 else project as a reference.

https://github.com/urish/angular2-moment/blob/master/package.json

which is more less what you propose, but maybe we could lower the versions a litttle bit?
What do you think about:

```
  "devDependencies": {
    "@types/core-js": "^0.9.36",
    "rimraf": "^2.6.1",
    "ts-loader": "^2.0.1",
    "typescript": "^2.1.6",
    "@angular/common": "4.4.6",
    "@angular/compiler": "4.4.6",
    "@angular/core": "4.4.6",
    "@angular/platform-browser": "4.4.6",
    "@angular/platform-browser-dynamic": "4.4.6",
    "rxjs": "^5.0.3",
    "zone.js": ">=0.8.5"
  },
  "dependencies": {
    "bootstrap": "^3.3",
    "eonasdan-bootstrap-datetimepicker": ">=4.17.22 <5.0.0",
    "@types/eonasdan-bootstrap-datetimepicker": "^4.17.22",
    "@types/jquery": "^2.0.33",
    "moment": "^2.10",
    "moment-timezone": "^0.4.0",
    "jquery": "^1.8.3 || ^2.0"
  },
  "peerDependencies": {
    "@angular/common": ">=2.0.0",
    "@angular/core": ">=2.0.0"
  }
```


maybe it would be a good idea to fix versions in `devDependencies` section.

Reply:
@atais do you want me to edit the devDependencies to lower version? to use version 2.0.0 for example? imo it doesnt matter..
 And I think @types/eonasdan-bootstrap-datetimepicker and @types/jquery should be in devDependecies too, but before changing these I'm asking you for something  else you want me to edit.